From patchwork Thu Sep 10 12:57:30 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Jo-Philipp Wich <jo@mein.io>
X-Patchwork-Id: 1361550
Return-Path: <openwrt-devel-bounces+incoming=patchwork.ozlabs.org@lists.openwrt.org>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=pass (sender SPF authorized) smtp.mailfrom=lists.openwrt.org
 (client-ip=2001:8b0:10b:1231::1; helo=merlin.infradead.org;
 envelope-from=openwrt-devel-bounces+incoming=patchwork.ozlabs.org@lists.openwrt.org;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
 dmarc=none (p=none dis=none) header.from=mein.io
Authentication-Results: ozlabs.org; dkim=pass (2048-bit key;
 secure) header.d=lists.infradead.org header.i=@lists.infradead.org
 header.a=rsa-sha256 header.s=merlin.20170209 header.b=sJMYOQPr;
 dkim-atps=neutral
Received: from merlin.infradead.org (merlin.infradead.org
 [IPv6:2001:8b0:10b:1231::1])
 (using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
 key-exchange X25519 server-signature RSA-PSS (4096 bits) server-digest
 SHA256) (No client certificate requested)
 by ozlabs.org (Postfix) with ESMTPS id 4BnJqm1xHCz9sTg
 for <incoming@patchwork.ozlabs.org>; Thu, 10 Sep 2020 22:59:15 +1000 (AEST)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
 d=lists.infradead.org; s=merlin.20170209; h=Sender:Content-Transfer-Encoding:
 Content-Type:Cc:List-Subscribe:List-Help:List-Post:List-Archive:
 List-Unsubscribe:List-Id:MIME-Version:Message-Id:Date:Subject:To:From:
 Reply-To:Content-ID:Content-Description:Resent-Date:Resent-From:Resent-Sender
 :Resent-To:Resent-Cc:Resent-Message-ID:In-Reply-To:References:List-Owner;
 bh=DiaZrg8lCKUOXYPqSUte9aioTF6y6obdvDUCF9X5vqo=; b=sJMYOQPr2aHF3PowujdwlcU8mZ
 HpL6WeuOLBzE7F4kglo91/F9F+a7ekLFqEetfLzg/6+m3sUkmdFk4lWYgJFOp3BI6lqmd2StZVa6j
 k4JhsjaHIvdbuNOcIm7fotTNF5y2DfCl52x54S8HR83NFpwA1U5hlpaW5aiuQpJylN7qzehQtemST
 kfey3mLDjG+xno3Q5ba5qh2WyO4b+W5IRKEnKn0KrIb6f/erA5ErfRMQkbRSCAd7EosJzaOXLLFrW
 KcmSPjGF3XiS0U8aMQV6U3fhj9iXjdGYAXYDBElyfo9IYz7wAfNfW/zYlXoruUOOjxIkCcocanaWV
 G12mjYCg==;
Received: from localhost ([::1] helo=merlin.infradead.org)
 by merlin.infradead.org with esmtp (Exim 4.92.3 #3 (Red Hat Linux))
 id 1kGM8w-0008RO-Jt; Thu, 10 Sep 2020 12:57:38 +0000
Received: from mxout01.bytecamp.net ([212.204.60.217])
 by merlin.infradead.org with esmtps (Exim 4.92.3 #3 (Red Hat Linux))
 id 1kGM8s-0008Q9-Cz
 for openwrt-devel@lists.openwrt.org; Thu, 10 Sep 2020 12:57:35 +0000
Received: by mxout01.bytecamp.net (Postfix, from userid 1001)
 id 0FFB57F9A2; Thu, 10 Sep 2020 14:57:33 +0200 (CEST)
Received: from mail.bytecamp.net (mail.bytecamp.net [212.204.60.9])
 by mxout01.bytecamp.net (Postfix) with ESMTP id C88CD7F99F
 for <openwrt-devel@lists.openwrt.org>; Thu, 10 Sep 2020 14:57:32 +0200 (CEST)
Received: (qmail 94879 invoked from network); 10 Sep 2020 14:57:32 +0200
Received: from unknown (HELO j7.lan) (jo%wwsnet.net@95.90.24.103)
 by mail.bytecamp.net with ESMTPS (DHE-RSA-AES128-GCM-SHA256 encrypted);
 10 Sep 2020 14:57:32 +0200
From: Jo-Philipp Wich <jo@mein.io>
To: openwrt-devel@lists.openwrt.org
Subject: [PATCH 1/2] fakeroot: make fakeroot script relocatable
Date: Thu, 10 Sep 2020 14:57:30 +0200
Message-Id: <20200910125731.3415057-1-jo@mein.io>
X-Mailer: git-send-email 2.28.0
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20200910_085734_582590_F81975C3 
X-CRM114-Status: GOOD (  11.12  )
X-Spam-Score: -0.7 (/)
X-Spam-Report: SpamAssassin version 3.4.4 on merlin.infradead.org summary:
 Content analysis details:   (-0.7 points)
 pts rule name              description
 ---- ----------------------
 --------------------------------------------------
 -0.7 RCVD_IN_DNSWL_LOW      RBL: Sender listed at https://www.dnswl.org/,
 low trust [212.204.60.217 listed in list.dnswl.org]
 0.0 SPF_HELO_NONE          SPF: HELO does not publish an SPF Record
 0.0 SPF_NONE               SPF: sender does not publish an SPF Record
X-BeenThere: openwrt-devel@lists.openwrt.org
X-Mailman-Version: 2.1.29
Precedence: list
List-Id: OpenWrt Development List <openwrt-devel.lists.openwrt.org>
List-Unsubscribe: <https://lists.openwrt.org/mailman/options/openwrt-devel>,
 <mailto:openwrt-devel-request@lists.openwrt.org?subject=unsubscribe>
List-Archive: <http://lists.openwrt.org/pipermail/openwrt-devel/>
List-Post: <mailto:openwrt-devel@lists.openwrt.org>
List-Help: <mailto:openwrt-devel-request@lists.openwrt.org?subject=help>
List-Subscribe: <https://lists.openwrt.org/mailman/listinfo/openwrt-devel>,
 <mailto:openwrt-devel-request@lists.openwrt.org?subject=subscribe>
Cc: Jo-Philipp Wich <jo@mein.io>, Daniel Golle <daniel@makrotopia.org>
Sender: "openwrt-devel" <openwrt-devel-bounces@lists.openwrt.org>
Errors-To: openwrt-devel-bounces+incoming=patchwork.ozlabs.org@lists.openwrt.org

Patch the fakeroot script template to discover faked and libfakeroot.so
relative to the STAGING_DIR_HOST environment variable, similar to how it
is done for automake, libtool, quilt and autoconf already.

This avoids the need for passing the paths to faked and libfakeroot.so
manually every time we invoke fakeroot and subsequently allows us to
drop OS X specific logic.

Signed-off-by: Jo-Philipp Wich <jo@mein.io>
---
 tools/fakeroot/patches/000-relocatable.patch | 25 ++++++++++++++++++++
 1 file changed, 25 insertions(+)
 create mode 100644 tools/fakeroot/patches/000-relocatable.patch

diff --git a/tools/fakeroot/patches/000-relocatable.patch b/tools/fakeroot/patches/000-relocatable.patch
new file mode 100644
index 0000000000..9f6915bfe8
--- /dev/null
+++ b/tools/fakeroot/patches/000-relocatable.patch
@@ -0,0 +1,25 @@
+--- a/scripts/fakeroot.in
++++ b/scripts/fakeroot.in
+@@ -30,12 +30,19 @@ fatal ()
+ }
+ 
+ # strip /bin/fakeroot to find install prefix
+-FAKEROOT_PREFIX=@prefix@
+-FAKEROOT_BINDIR=@bindir@
++if [ -n "$STAGING_DIR_HOST" ]; then
++    FAKEROOT_PREFIX="${STAGING_DIR_HOST}"
++    FAKEROOT_BINDIR="${STAGING_DIR_HOST}/bin"
++    FAKEROOT_LIBDIR="${STAGING_DIR_HOST}/lib"
++else
++    FAKEROOT_PREFIX=@prefix@
++    FAKEROOT_BINDIR=@bindir@
++    FAKEROOT_LIBDIR=@libdir@
++fi
+ 
+ USEABSLIBPATH=@LDPRELOADABS@
+ LIB=lib@fakeroot_transformed@@DLSUFFIX@
+-PATHS=@libdir@:${FAKEROOT_PREFIX}/lib64/libfakeroot:${FAKEROOT_PREFIX}/lib32/libfakeroot
++PATHS=${FAKEROOT_LIBDIR}:${FAKEROOT_PREFIX}/lib64/libfakeroot:${FAKEROOT_PREFIX}/lib32/libfakeroot
+ FAKED=${FAKEROOT_BINDIR}/@faked_transformed@
+ 
+ FAKED_MODE="unknown-is-root"

From patchwork Thu Sep 10 12:57:31 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Jo-Philipp Wich <jo@mein.io>
X-Patchwork-Id: 1361551
Return-Path: <openwrt-devel-bounces+incoming=patchwork.ozlabs.org@lists.openwrt.org>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=pass (sender SPF authorized) smtp.mailfrom=lists.openwrt.org
 (client-ip=2001:8b0:10b:1231::1; helo=merlin.infradead.org;
 envelope-from=openwrt-devel-bounces+incoming=patchwork.ozlabs.org@lists.openwrt.org;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
 dmarc=none (p=none dis=none) header.from=mein.io
Authentication-Results: ozlabs.org; dkim=pass (2048-bit key;
 secure) header.d=lists.infradead.org header.i=@lists.infradead.org
 header.a=rsa-sha256 header.s=merlin.20170209 header.b=sSzCIkFr;
 dkim-atps=neutral
Received: from merlin.infradead.org (merlin.infradead.org
 [IPv6:2001:8b0:10b:1231::1])
 (using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
 key-exchange X25519 server-signature RSA-PSS (4096 bits) server-digest
 SHA256) (No client certificate requested)
 by ozlabs.org (Postfix) with ESMTPS id 4BnJqn3SRxz9sTr
 for <incoming@patchwork.ozlabs.org>; Thu, 10 Sep 2020 22:59:17 +1000 (AEST)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
 d=lists.infradead.org; s=merlin.20170209; h=Sender:Content-Transfer-Encoding:
 Content-Type:Cc:List-Subscribe:List-Help:List-Post:List-Archive:
 List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:Message-Id:Date:
 Subject:To:From:Reply-To:Content-ID:Content-Description:Resent-Date:
 Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:List-Owner;
 bh=q2iaWQ0R6v9mRC3LGGConFyRvPAQgOc4fGzlPIPyTHI=; b=sSzCIkFrw3SOG0CEwljMdA1yi
 8pO/3craqsi2Yn4xiPcEqHLPuPSsGPWVuiEJ0jXvNObQ/JLu09mmvT003o+QI6I79oQN/r3nEMOPB
 hfNMXXUlGWAyiRd296kM6I2RZPUX5jTvuakc3bFsKgqa/KBRv3wcGxBHSV+gzIoEs+enTyqXMstvq
 xKj6/P3C296cFGDsC6j8sRWvx+HUo1sMNY+jREbG/oMoZXGF64QN13aAkjoL0aMpp1ZMahZKqeOjB
 00PTCSty6qcoUl/ssW5q/MFppz2cLHK14kCFrGbcz+3be3pJyzaxgsSYHPe6A0xrivKBhewv6Z6ls
 xmecgTaLw==;
Received: from localhost ([::1] helo=merlin.infradead.org)
 by merlin.infradead.org with esmtp (Exim 4.92.3 #3 (Red Hat Linux))
 id 1kGM8v-0008RB-JX; Thu, 10 Sep 2020 12:57:37 +0000
Received: from mxout01.bytecamp.net ([212.204.60.217])
 by merlin.infradead.org with esmtps (Exim 4.92.3 #3 (Red Hat Linux))
 id 1kGM8s-0008QB-Cz
 for openwrt-devel@lists.openwrt.org; Thu, 10 Sep 2020 12:57:35 +0000
Received: by mxout01.bytecamp.net (Postfix, from userid 1001)
 id 2D9787F9A5; Thu, 10 Sep 2020 14:57:33 +0200 (CEST)
Received: from mail.bytecamp.net (mail.bytecamp.net [212.204.60.9])
 by mxout01.bytecamp.net (Postfix) with ESMTP id EB2AD7F9A0
 for <openwrt-devel@lists.openwrt.org>; Thu, 10 Sep 2020 14:57:32 +0200 (CEST)
Received: (qmail 94894 invoked from network); 10 Sep 2020 14:57:32 +0200
Received: from unknown (HELO j7.lan) (jo%wwsnet.net@95.90.24.103)
 by mail.bytecamp.net with ESMTPS (DHE-RSA-AES128-GCM-SHA256 encrypted);
 10 Sep 2020 14:57:32 +0200
From: Jo-Philipp Wich <jo@mein.io>
To: openwrt-devel@lists.openwrt.org
Subject: [PATCH 2/2] rules.mk: simplify FAKEROOT command line
Date: Thu, 10 Sep 2020 14:57:31 +0200
Message-Id: <20200910125731.3415057-2-jo@mein.io>
X-Mailer: git-send-email 2.28.0
In-Reply-To: <20200910125731.3415057-1-jo@mein.io>
References: <20200910125731.3415057-1-jo@mein.io>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20200910_085734_554766_FB8C0DE1 
X-CRM114-Status: UNSURE (   9.25  )
X-CRM114-Notice: Please train this message.
X-Spam-Score: -0.7 (/)
X-Spam-Report: SpamAssassin version 3.4.4 on merlin.infradead.org summary:
 Content analysis details:   (-0.7 points)
 pts rule name              description
 ---- ----------------------
 --------------------------------------------------
 -0.7 RCVD_IN_DNSWL_LOW      RBL: Sender listed at https://www.dnswl.org/,
 low trust [212.204.60.217 listed in list.dnswl.org]
 0.0 SPF_HELO_NONE          SPF: HELO does not publish an SPF Record
 0.0 SPF_NONE               SPF: sender does not publish an SPF Record
X-BeenThere: openwrt-devel@lists.openwrt.org
X-Mailman-Version: 2.1.29
Precedence: list
List-Id: OpenWrt Development List <openwrt-devel.lists.openwrt.org>
List-Unsubscribe: <https://lists.openwrt.org/mailman/options/openwrt-devel>,
 <mailto:openwrt-devel-request@lists.openwrt.org?subject=unsubscribe>
List-Archive: <http://lists.openwrt.org/pipermail/openwrt-devel/>
List-Post: <mailto:openwrt-devel@lists.openwrt.org>
List-Help: <mailto:openwrt-devel-request@lists.openwrt.org?subject=help>
List-Subscribe: <https://lists.openwrt.org/mailman/listinfo/openwrt-devel>,
 <mailto:openwrt-devel-request@lists.openwrt.org?subject=subscribe>
Cc: Jo-Philipp Wich <jo@mein.io>, Daniel Golle <daniel@makrotopia.org>
Sender: "openwrt-devel" <openwrt-devel-bounces@lists.openwrt.org>
Errors-To: openwrt-devel-bounces+incoming=patchwork.ozlabs.org@lists.openwrt.org

Since fakeroot is patched to discover related ressources relative to the
STAGING_DIR_HOST environment variable, there is no need to pass the path
to faked or the preload library manually anymore.

Signed-off-by: Jo-Philipp Wich <jo@mein.io>
---
 rules.mk | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/rules.mk b/rules.mk
index 8b2424f2ba..3214395e1f 100644
--- a/rules.mk
+++ b/rules.mk
@@ -264,12 +264,7 @@ endif
 
 BUILD_KEY=$(TOPDIR)/key-build
 
-ifeq ($(HOST_OS),Darwin)
-  FAKEROOT_SO:=$(STAGING_DIR_HOST)/lib/libfakeroot.dylib
-else
-  FAKEROOT_SO:=$(STAGING_DIR_HOST)/lib/libfakeroot.so
-endif
-FAKEROOT:=$(STAGING_DIR_HOST)/bin/fakeroot -l $(FAKEROOT_SO) -f $(STAGING_DIR_HOST)/bin/faked
+FAKEROOT:=$(STAGING_DIR_HOST)/bin/fakeroot
 
 TARGET_CC:=$(TARGET_CROSS)gcc
 TARGET_CXX:=$(TARGET_CROSS)g++
